modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3282,10 +3299,15 @@ void GDScriptAnalyzer::reduce_call(GDScriptParser::CallNode *p_call, bool p_is_a
 
 	bool is_constructor = (base_type.is_meta_type || (p_call->callee && p_call->callee->type == GDScriptParser::Node::IDENTIFIER)) && p_call->function_name == SNAME("new");
 
-	if (is_constructor && Engine::get_singleton()->has_singleton(base_type.native_type)) {
-		push_error(vformat(R"(Cannot construct native class "%s" because it is an engine singleton.)", base_type.native_type), p_call);
-		p_call->set_datatype(call_type);
-		return;
+	if (is_constructor) {
+		if (Engine::get_singleton()->has_singleton(base_type.native_type)) {
+			push_error(vformat(R"(Cannot construct native class "%s" because it is an engine singleton.)", base_type.native_type), p_call);
+			p_call->set_datatype(call_type);
+			return;
+		}
+		if (_is_type_abstract_class(&base_type)) {
+			push_error(vformat(R"*(Cannot call constructor on abstract class "%s".)*", base_type.to_string()), p_call);
```

Comment:
```suggestion
			push_error(vformat(R"(Cannot construct abstract class "%s".)", base_type.to_string()), p_call);
```

In my opinion it looks better and more consistent. Let me know if you have other options.